aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -304,15 +304,12 @@ void index_copy_kernel(
 
 template <typename scalar_t, typename mask_t>
 void cpu_masked_fill_kernel(TensorIterator& iter, scalar_t value) {
-  auto is_mask_bool = std::is_same<mask_t, bool>::value;
   auto loop = [&](char** data, const int64_t* strides, int64_t n) {
     char* dst = data[0];
     char* mask = data[1];
     for (const auto i : c10::irange(n)) {
       mask_t mask_value = *(mask_t*)(mask + strides[1] * i);
```

Comment:
Nit (as mask type is always bool
```suggestion
      auto mask_value = *static_cast<bool*>(mask + strides[1] * i);
```